### PR TITLE
jwt-cpp/0.3.1 attempted to add patches

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **jwt-cpp/0.3.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

I tried to add some patches ~however I still need some training wheels... It's not able to find my patches...~ I followed the fmt recipe 

<details><summary> Old error </summary>

```
ERROR: jwt-cpp/0.3.1: Error in build() method, line 32
        tools.patch(**patch)
        FileNotFoundError: [Errno 2] No such file or directory: 'patches/0001-fix-windows-warning.patch'
```

</details>

### CODE REVIEW

File: recipes/jwt-cpp/all/conanfile.py [hunk 1]
```diff
@@ -18,15 +18,22 @@ def _source_subfolder(self):
...
+        self.requires("openssl/1.1.1f")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
```
Comment (prince-chrismc) [inline recipes/jwt-cpp/all/conanfile.py:30]: @uilianries could you be so kind to let me know if this is correct for a header-only recipe?

**Comment by uilianries** [comment]
Why do you need that patch? Where did you find it?

**Comment by prince-chrismc** [comment]
The openssl version is from here https://github.com/Thalhammer/jwt-cpp/issues/57

Warning from my windows CI
```1>c:\users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\base.h(131): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data
1>c:\users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\base.h(132): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data
1>c:\users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\base.h(133): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data
1>c:\users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\base.h(134): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data
1>c:\users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\base.h(150): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(81): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(102): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(105): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(118): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(170): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(171): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(283): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(322): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(325): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(336): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(358): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(394): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(433): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(435): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>C:\Users\cmcarthu\.conan\data\jwt-cpp\0.3.1\_\_\package\2e93e967200138a42daa20b0593bc70269764eb7\include\jwt.h(527): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
```

File: recipes/jwt-cpp/all/conanfile.py [hunk 2]
```diff
@@ -8,8 +8,8 @@ class JwtCppConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/Thalhammer/jwt-cpp"
     description = "A C++ JSON Web Token library for encoding/decoding"
-    topics = ("jwt-cpp", "json", "jwt", "header-only", "conan-recipe")
-
+    topics = ("jwt-cpp", "json", "jwt", "header-only", "jose")
```
Comment (uilianries) [inline recipes/jwt-cpp/all/conanfile.py:11]: what does mean "jose" ?

File: recipes/jwt-cpp/all/patches/0001-fix-windows-warning.patch [hunk 3]
```diff
@@ -0,0 +1,69 @@
+diff --git a/include/jwt-cpp/base.h b/include/jwt-cpp/base.h
```
Comment (uilianries) [inline recipes/jwt-cpp/all/patches/0001-fix-windows-warning.patch:1]: Remove this patch and apply that one related to OpenSSL. Warnings are not errors and it's part of the upstream. If something is wrong, report to the upstream first.

File: recipes/jwt-cpp/all/conanfile.py (hunk 2 above)
Comment (prince-chrismc) [reply recipes/jwt-cpp/all/conanfile.py:11]: json object signing and encryption
https://tools.ietf.org/html/rfc7520

File: recipes/jwt-cpp/all/patches/0001-fix-windows-warning.patch (hunk 3 above)
Comment (prince-chrismc) [reply recipes/jwt-cpp/all/patches/0001-fix-windows-warning.patch:1]: Alrighty, yeah I got a PR trying to fix them.

File: recipes/jwt-cpp/all/conanfile.py (hunk 2 above)
Comment (uilianries) [reply recipes/jwt-cpp/all/conanfile.py:11]: nice! I didn't know about, thanks!

**Pull request merged**